src/boot/efi/meson.build

```diff
@@ -244,13 +243,23 @@ if have_gnu_efi
                        '--build-id=sha1',
                        '-L', efi_libdir,
                        efi_crt0]
+        efi_staticlib = []
         if ['aarch64', 'arm', 'riscv64'].contains(efi_arch)
                 # Aarch64, ARM32 and 64bit RISC-V don't have an EFI capable objcopy.
                 # Use 'binary' instead, and add required symbols manually.
                 efi_ldflags += ['--defsym=EFI_SUBSYSTEM=0xa']
                 efi_format = ['-O', 'binary']
+
+                stub_sources += 'linux-aarch64.c'
+                stub_sources += 'stdclib.c'
+
+                # libfdt needed for arm to pass initrd to kernel
+                cc.find_library('fdt')
+                efi_staticlib += ['/usr/lib/@0@-linux-gnu/libfdt.a'.format(efi_arch)]
```

Comment:
can't they provide a pkg-config file with the right path? this is otherwise going to be really hard to get right for the various distros

Reply:
I can confirm that neither Ubuntu or Debian have pkg-config files in the `libfdt-dev` package (same as Arch as pointed out by @DaanDeMeyer).

Reply:
Yeah this is completely broken - it will break cross builds, SDK builds (yocto), etc. Please fix it and avoid hard coded paths.